ocs_ci/utility/pagerduty.py

```diff
@@ -0,0 +1,336 @@
+            }
+        }
+
+    def get_integration_dict(self, vendor_name):
+        """
+        Get a structure prepared to be a payload for integration creation via API.
+
+        Args:
+            vendor_name (str): Name of vendor that is used in integration
+
+        Returns:
+            dict: Structure containing all data required to by PagerDuty API
+                in order to create an integration
+
+        """
+
+        vendor_id = self.get_vendor_id("Prometheus")
+        return {
+            "integration": {
+                "type": "generic_events_api_inbound_integration",
+                "name": "Prometheus",
+                "vendor": {"type": "vendor_reference", "id": vendor_id},
+            }
+        }
+
+    def wait_for_incident_cleared(self, summary, timeout=1200, sleep=5):
+        """
+        Search for incident to be cleared.
+
+        Args:
+            summary (str): Incident summary
+            sleep (int): Number of seconds to sleep in between incidents search
+
+        Returns:
+            list: List of incident records
+
+        """
+        while timeout > 0:
+            incidents_response = self.get(
+                "incidents",
+            )
+            msg = f"Request {incidents_response.request.url} failed"
+            assert incidents_response.ok, msg
+            incidents = [
+                incident
+                for incident in incidents_response.json().get("incidents")
+                if summary in incident.get("summary")
+                and incident.get("status") != "resolved"
+            ]
+            logger.info(
+                f"Checking for {summary} incidents. There should be no incidents ... "
+                f"{len(incidents)} found"
+            )
+            if len(incidents) == 0:
+                break
+            time.sleep(sleep)
+            timeout -= sleep
+        return incidents
+
+    def check_incident_cleared(self, summary, measure_end_time, time_min=120):
+        """
+        Check that all incidents with provided summary are cleared.
+
+        Args:
+            summary (str): Incident summary
+            measure_end_time (int): Timestamp of measurement end
+            time_min (int): Number of seconds to wait for incidents to be cleared
+                since measurement end
+
+        """
+        time_actual = time.time()
+        time_wait = int((measure_end_time + time_min) - time_actual)
+        if time_wait > 0:
+            logger.info(
+                f"Waiting for approximately {time_wait} seconds for incidents "
+                f"to be cleared ({time_min} seconds since measurement end)"
+            )
+        else:
+            time_wait = 1
+        cleared_incidents = self.wait_for_incident_cleared(
+            summary=summary, timeout=time_wait
+        )
+        logger.info(f"Cleared incidents: {cleared_incidents}")
+        assert len(cleared_incidents) == 0, f"{summary} incidents were not cleared"
```

Comment:
I would rather avoid asserts in a module. An exception would be a better choice here.